.travis.yml

```diff
@@ -54,6 +54,12 @@ matrix:
       os: linux
       compiler: gcc
       before_script:
+        # At least one of jobs should use a specific version of cmake so that we know the lower bound version
+        - cmake --version
+        - sudo rm `which cmake`
+        - sh -c "wget https://cmake.org/files/v3.12/cmake-3.12.4.tar.gz && tar -xf cmake-3.* && cd cmake-3.12.4 && ./bootstrap && make -j4 && sudo make install"
```

Comment:
Unfortunately, 3.12.4 is the lowest common denominator. Although, if we look at our dependencies, the highest requirement is 3.0. But, for whatever reason, it couldn't find OpenSSL properly when I used 3.0.